### ISSUE TITLE
Empty YAML config file causes IllegalArgumentException when loading

### ISSUE THREAD

**Issue by Prof-Bloodstone** [uchuhimo/konf#59]
I was trying to use konf to manage my config files and migrations, together with default config.

It looks like konf can't handle, when file:

- doesn't exist
- exists, but is empty

The stacktrace for empty YAML file:
```
Exception in thread "main" java.lang.IllegalArgumentException: Parameter specified as non-null is null: method com.uchuhimo.konf.source.yaml.YamlConstructor$1.construct, parameter node
	at com.uchuhimo.konf.source.yaml.YamlConstructor$1.construct(YamlProvider.kt)
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:151)
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:525)
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:453)
	at com.uchuhimo.konf.source.yaml.YamlProvider.inputStream(YamlProvider.kt:45)
	at com.uchuhimo.konf.source.Provider$DefaultImpls.file(Provider.kt:84)
	at com.uchuhimo.konf.source.yaml.YamlProvider.file(YamlProvider.kt:37)
	at com.uchuhimo.konf.source.Loader.file(Loader.kt:81)
	at com.uchuhimo.konf.source.Loader.file$default(Loader.kt:80)
	at dev.bloodstone.konf_test.MainKt.main(Main.kt:24)
```

Adding `dummy-value:` to config, causes it to load.

I don't know how important is handling of such edge-case, since it looks like konf is not made for my use case, and such things shouldn't happen under normal circuimstances. But I thought I'd report it anyway.

**Comment by benjishults** [comment]
I'm facing this problem, too.  In my case, the file is empty.

**Comment by uchuhimo** [comment]
@Prof-Bloodstone @benjishults I have released a new version v0.23.0 that supports to load empty YAML/JSON files. You can try it now. Any further feedback is welcome.

In the case of absence of file, you can use `config.from.file(path, optional=true)` or `config.enable(Feature.OPTIONAL_SOURCE_BY_DEFAULT)`.
